### DISCUSSION TITLE
Image format and file extension

### DISCUSSION THREAD

**Discussion by herb52** [ImageMagick/ImageMagick#7148] [Development]
### ImageMagick version

7.1.1-29

### Operating system, version and so on

Windows 10

### Description

Hello,

How does ImageMagick determine the format of an image:
- is it done only by extension of image file or
- does it read the header of a file

Background to this question is:
I work on an application that reads and modifies metadata with ExifTool.
WIth ImageMagick I create thumbnails e.g. for *.avif and/or *.heic images.
ExifTool creates backup-file by adding _original to the filename.
ImageMagick works fine for a file image.avif, but it does not work for the backup file image.avif_original

Thanks for your answer in advance
Best regards
herb

**Comment by fmw42** [comment]
Why don't use put the _original as image_original.avif?

**Comment by snibgo** [comment]
IM reads file headers.

IM can correctly read a file named `x.avif_original`:
```
magick x.avif_original info:
x.avif_original AVIF 267x233 267x233+0+0 12-bit sRGB 2910B 0.000u 0:00.000
```

> but it does not work for the backup file image.avif_original

Please show the command you used, any error or warning text output, and image.avif_original, eg zip it and post it here.

**Comment by herb52** [comment]
Hello,

thanks for your comments and help.
The question is answered in meantime. A colleague has shown to me that ImageMagick uses the file header to determine the image format.

Best regards
herb
